src/test/java/it/cnr/isti/hpc/wikipedia/article/ArticleTest.java

```diff
@@ -124,13 +124,39 @@ public void testExtractingLinksOtherLang(){
 	}
 
 	/*
+	* Test for inline ref parsing.
+	* */
+	@Test
+	public void testReferences() throws IOException {
+		ArticleParser parser = new ArticleParser(Language.EN);
+		Article a = new Article();
+		String mediawiki = IOUtils.getFileAsUTF8String("./src/test/resources/misc/anarchism.xml");
+		parser.parse(a, mediawiki);
+
+		// check that the first paragraph of anarchism has 10 refs detected.
+		List<Ref> refs = a.getParagraphsWithLinks().get(0).getRefs();
+		assertEquals(10, refs.size());
+
+		// Check first and last detected references inside values
+		Ref fistRef = refs.get(0);
+		assertEquals(157, fistRef.getStart());
+		assertEquals(435, fistRef.getEnd());
+		assertNotEquals(-1, fistRef.getText().indexOf("a social philosophy that rejects authoritarian government and maintains"));
```

Comment:
This reads a bit funny, like a double negative or something.
What about
```
assertTrue(firstRef.contains("a social philosophy that rejects authoritarian government and maintains"))
```
?

Reply:
You could also refactor these 4 lines into a helper method like
```
checkRef(ref, start, end, text) {
    assertEquals(start, ref.getStart());
    assertEquals(end, ref.getEnd());
    assertTrue(ref.contains(text))
}
```
which would allow people to extend the tests more easily in future.